### PR TITLE
Handle unknown zip signature 0x8074b50

### PR DESCRIPTION
**Pull request by ghost**
## Issue
We have multiple client zip files in a format that the original node-unzipper library does not handle.  These zip files contain data chunk signatures of 0x8074b50.  In the original "ZJONSSON/node-unzipper" library, this threw an "unknown signature" error after extracting a single file.  This signature **may** be a different format of the "data descriptor" chunk located between each file in the zip, as the normal data descriptor signature is 0x08074b50 vs 0x8074b50.

## Description of change
This change handles this specific signature and returns the size of the chunk to move past it in the stream processing.  In order for these chunks to be processed properly, the file size check was changed from:
`var fileSizeKnown = !(vars.flags & 0x08) || vars.compressedSize > 0,
            eof;`
To:
`var fileSizeKnown = !(vars.flags & 0x08);`

This change matches the original repo that "ZJONSSON/node-unzipper" was forked from named "EvanOxfeld/node-unzip".  All of the other forks of "EvanOxfeld/node-unzip" or "ZJONSSON/node-unzipper" (that I've found) use this code `var fileSizeKnown = !(vars.flags & 0x08);` to determine if file size is known.  This change allows these chunks to be handled as data descriptors, rather than files.

One effect of these changes was that the code no longer threw an error for password protected zip files.  The original "ZJONSSON/node-unzipper" library actually does handle password protected zip files, but not within the Parse() method.  I added a specific check (found in other forks of "EvanOxfeld/node-unzip") to throw an error in the Parse method when a password protected zip is encountered.

### CODE REVIEW

File: lib/parse.js [hunk 1]
```diff
@@ -50,6 +50,9 @@ Parse.prototype._readRecord = function () {
     if (signature === 0x04034b50) {
       return self._readFile();
     }
+    else if (signature === 0x8074b50) {
+      return data.length;
+    }
```
Comment (matthewsecrist) [inline lib/parse.js:55]: I'm not super familiar with the internals of how this works. Why does returning the length of the data work for this?

File: lib/parse.js [hunk 2]
```diff
@@ -184,14 +193,19 @@ Parse.prototype._readFile = function () {
...
-            .on('finish', function() {
-              return fileSizeKnown ?
-                self._readRecord().then(resolve).catch(reject) :
-                self._processDataDescriptor(entry).then(resolve).catch(reject);
-            });
+          .pipe(inflater)
+          .on('pause', function() {
+            return fileSizeKnown ?
+            self._readRecord().then(resolve).catch(reject) :
+            self._processDataDescriptor(entry).then(resolve).catch(reject);
```
Comment (matthewsecrist) [inline lib/parse.js]:
```suggestion
            return fileSizeKnown
              ? self._readRecord().then(resolve).catch(reject)
              : self._processDataDescriptor(entry).then(resolve).catch(reject);
```

File: lib/parse.js [hunk 3]
```diff
@@ -184,14 +193,19 @@ Parse.prototype._readFile = function () {
...
+          })
+          .on('error',function(err) {self.emit('error',err);})
+          .pipe(entry)
+          .on('finish', function() {
+            return fileSizeKnown ?
+              self._readRecord().then(resolve).catch(reject) :
+              self._processDataDescriptor(entry).then(resolve).catch(reject);
```
Comment (matthewsecrist) [inline lib/parse.js:202]:
```suggestion
            return fileSizeKnown
              ? self._readRecord().then(resolve).catch(reject)
              : self._processDataDescriptor(entry).then(resolve).catch(reject);
```

File: lib/parse.js (hunk 1 above)
Comment (ghost) [reply lib/parse.js:55]: Basically, it moves us past this chunk in the stream.

File: lib/parse.js (hunk 3 above)
Comment (ghost) [reply lib/parse.js:202]: @matthewsecrist  I can make these format changes if you feel strongly, I was just trying to match the code in the original lib (which doesn't follow our eslint rules at all obviously)?

Comment (matthewsecrist) [reply lib/parse.js:202]: I don't mind if we don't make these adjustments. Just figured I'd make the suggestion. :)

**Review by matthewsecrist — APPROVED** [review]

File: lib/parse.js (hunk 1 above)
Comment (pklingem) [reply lib/parse.js:55]: how does it move us past this chunk in the stream?

File: lib/parse.js (hunk 2 above)
Comment (pklingem) [reply lib/parse.js]: based on the comment below I know you're trying to match the style in this repo, even if you don't put the `?` or `:` in front, there is an indentation problem here that doesn't exist elsewhere, the two lines after `return fileSizeUnknown ?` need to be indented 2 spaces.

**Review by pklingem — CHANGES_REQUESTED** [review]
I'd like to see a PR on `rise-zip-extractor` with this in it, you can use [this process](https://github.com/articulate/react-components/blob/master/CONTRIBUTING.md#testing-and-packaging-changes). The benefit is: I wouldn't want to merge this PR until we're sure it's addressed the downstream problem.

Beyond that, there are some questions about _how_ this solves the problem in the comments above that we should understand.

**Review by pklingem — APPROVED** [review]
looking good! Thank you for addressing my concerns :ship::it:

**Pull request merged**